addons/sale_timesheet/tests/test_sale_service.py

```diff
@@ -89,6 +89,8 @@ def test_sale_service(self):
 
         self.assertEqual(self.sale_order.tasks_count, 2, "Adding a new service line on a confirmer SO should create a new task.")
 
+        # timesheets must be unlinked first before unlinking a task
+        task.timesheet_ids.unlink()
```

Comment:
That's a weird and risky "fix", since the timesheets are somewhat related to the linked SOL.
Moreover, `sale_timesheet` depends on `hr_timesheet`, so the first one should be aware of the second one...

Reply:
Ok, nevermind. But let's make the comment clearer:
```suggestion
        # delete timesheets before deleting the task, so as to trigger the error
        # about linked sales order lines and not the one about linked timesheets
        task.timesheet_ids.unlink()
```